Source/EventFlow.PostgreSql/EventStores/PostgresSqlEventPersistence.cs

```diff
@@ -0,0 +1,217 @@
+                    GlobalSequenceNumber ASC;";
+            var eventDataModels = await _connection.QueryAsync<EventDataModel>(
+                Label.Named("postgresql-fetch-events"),
+                cancellationToken,
+                sql,
+                new
+                    {
+                        FromId = startPosition,
+                        ToId = endPosition,
+                    })
+                .ConfigureAwait(false);
+
+            var nextPosition = eventDataModels.Any()
+                ? eventDataModels.Max(e => e.GlobalSequenceNumber) + 1
+                : startPosition;
+
+            return new AllCommittedEventsPage(new GlobalPosition(nextPosition.ToString()), eventDataModels);
+        }
+
+        public async Task<IReadOnlyCollection<ICommittedDomainEvent>> CommitEventsAsync(
+            IIdentity id,
+            IReadOnlyCollection<SerializedEvent> serializedEvents,
+            CancellationToken cancellationToken)
+        {
+            if (!serializedEvents.Any())
+            {
+                return new ICommittedDomainEvent[] { };
+            }
+
+            var eventDataModels = serializedEvents
+                .Select((e, i) => new EventDataModel
+                    {
+                        AggregateId = id.Value,
+                        AggregateName = e.Metadata[MetadataKeys.AggregateName],
+                        BatchId = Guid.Parse(e.Metadata[MetadataKeys.BatchId]),
+                        Data = e.SerializedData,
+                        Metadata = e.SerializedMetadata,
+                        AggregateSequenceNumber = e.AggregateSequenceNumber,
+                    })
+                .ToList();
+
+            _log.Verbose(
+                "Committing {0} events to POSTGRESQL event store for entity with ID '{1}'",
+                eventDataModels.Count,
+                id);
+
+            const string sql = @"
+                INSERT INTO
+                    EventFlow
+                        (BatchId, AggregateId, AggregateName, Data, Metadata, AggregateSequenceNumber)
+                    VALUES
+                        (@BatchId, @AggregateId, @AggregateName, @Data, @Metadata, @AggregateSequenceNumber);
+                SELECT lastval() FROM EventFlow;";
+
+            IReadOnlyCollection<long> ids;
+
+            try
+            {
+                ids = await _connection.InsertMultipleAsync<long, EventDataModel>(
+                    Label.Named("sqlite-insert-events"),
+                    cancellationToken,
+                    sql,
+                    eventDataModels)
+                    .ConfigureAwait(false);
+            }
+            catch (PostgresException e)
+            {
+                //https://www.postgresql.org/docs/9.4/static/errcodes-appendix.html
+                //Explanation of Error code
+                if (e.SqlState == "23505") //unique_violation
+                {
+                    throw new OptimisticConcurrencyException(e.ToString(), e);
```

Comment:
Remove the `e.ToString()` and use `e.Message` or write some alternative message that provides additional details.